scheduler/scheduler.py

```diff
@@ -725,35 +724,34 @@ def _assign_workers_to_job(self, job_id, scale_factor, worker_type,
           worker_type: The worker type to allocate.
           worker_state: A dict comprised of the following information:
             worker_ids: Worker IDs organized into servers.
-            reserved_worker_ids: A set of worker IDs that are already in use.
+            assigned_worker_ids: The set of worker IDs assigned so far.
             server_id_ptr: The server to assign workers from.
-            num_workers_assigned: The total number of allocated workers.
-          worker_assignments: A list of (job_id, worker_ids) assignment tuples.
-          worker_ids_for_job: An optional list of worker IDs to assign.
+          worker_assignments: A map from job_id to assigned worker_ids tuple.
         """
         worker_ids = worker_state['worker_ids']
-        reserved_worker_ids = worker_state['reserved_worker_ids']
+        assigned_worker_ids = worker_state['assigned_worker_ids']
         server_id_ptr = worker_state['server_id_ptr']
-        num_workers_assigned = worker_state['num_workers_assigned']
 
-        if worker_ids_for_job is None:
+        if job_id in worker_assignments:
+            worker_ids_for_job = list(worker_assignments[job_id])
+        else:
             worker_ids_for_job = []
-            while (len(worker_ids_for_job) < scale_factor and
```

Comment:
Add comment here about what this branch is doing (previous round worker assignments?)

Reply:
Wait which branch? This code is just assigning workers to the job - there may already be a set of assigned workers for the job but the only change in behavior in this function is what the initial value of of `worker_ids_for_job` will be

Reply:
Why is this function being called if there's already an assignment of workers for this job?

Reply:
The block of code at the end under the `if self._simulate` still needs to run